src/main/java/com/rackspace/salus/policy/manage/services/TenantManagement.java

```
@@ -0,0 +1,99 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspace.salus.policy.manage.services;
+
+import com.rackspace.salus.policy.manage.entities.TenantMetadata;
+import com.rackspace.salus.policy.manage.repositories.TenantMetadataRepository;
+import com.rackspace.salus.policy.manage.web.model.TenantMetadataCU;
+import com.rackspace.salus.telemetry.model.NotFoundException;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TenantManagement {
+
+  private final TenantMetadataRepository tenantMetadataRepository;
+
+  @Autowired
+  public TenantManagement(
+      TenantMetadataRepository tenantMetadataRepository) {
+    this.tenantMetadataRepository = tenantMetadataRepository;
+  }
+
+  /**
+   * Gets all known tenant metadata for a single tenant.
+   * @param tenantId The tenant to get the info for.
+   * @return The full TenantMetadata object.
+   */
+  public Optional<TenantMetadata> getMetadata(String tenantId) {
+    return tenantMetadataRepository.findByTenantId(tenantId);
+  }
+
+  /**
+   * Get the account type value for a tenant if it is set.
+   *
+   * @param tenantId The tenant to lookup.
+   * @return The accountType value for the tenant if it exists, otherwise null.
+   */
+  public String getAccountTypeByTenant(String tenantId) {
+    Optional<TenantMetadata> metadata = getMetadata(tenantId);
+    if (metadata.isEmpty()) {
+      return null;
```

or should it throw a not found?

---

Maybe?  Some of this logic I follows what we do in other places, like in API clients where we return `null` if something is not found.

If I change this I'd like to make it a little more consistent everywhere.

---

Using NotFound would make it easier to bundle this into the API if we wanted.  Right now it's only used locally so `null` works out.